src/platform/backends/libvirt/libvirt_virtual_machine.cpp

```diff
@@ -380,17 +388,14 @@ void mp::LibVirtVirtualMachine::ensure_vm_is_running()
 {
     std::lock_guard<decltype(state_mutex)> lock{state_mutex};
     auto domain = domain_by_name_for(vm_name, open_libvirt_connection().get());
-    auto present_state = domain_state_for(domain.get());
-    if (present_state != VirtualMachine::State::running)
+    if (!domain_is_running(domain.get()))
     {
         // Have to set 'off' here so there is an actual state change to compare to for
         // the cond var's predicate
         state = State::off;
         state_wait.notify_all();
-        if (present_state == VirtualMachine::State::unknown)
-            throw mp::StartException(vm_name, "Unable to determine state");
 
-        throw mp::StartException(vm_name, "Instance shutdown during start");
+        throw mp::StartException(vm_name, "Instance shut down while starting");
```

Comment:
I dunno, maybe "failed to start" ? Since a crash would result in this too, no?

Reply:
Yeah, I suppose being more generic here will be better.

Reply:
Done